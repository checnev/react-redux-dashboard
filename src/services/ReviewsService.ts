import moment from 'moment';
import type { IReview } from 'components/Review/index';

class ReviewsService {
  private static reviews: IReview[] = [
    {
      id: 1,
      author: 'Naruto Uzumaki',
      rating: 4.5,
      text: `Все хорошо, но в день - один, ну два подвига, не больше. <br/>
      Не бывает безвыходных ситуаций. Бывают ситуации, выход из которых тебя не устраивает.<br/>
      Никогда не сдаваться Встать, когда все рухнуло — вот настоящая сила.`,
      date: moment().add(-7, 'd'),
    },
    {
      id: 2,
      author: 'Sasuke Uchiha',
      rating: 5,
      text: `У тебя была мечта стать Хокаге? Если у тебя есть время бегать за мной, может, стоит больше тренироваться...<br/>
      А, Наруто?..<br/>
      Разве может тот, кто не спас товарища стать Хокаге?`,
      date: moment().add(-65, 'd'),
    },
    {
      id: 3,
      author: 'Kakashi Hatake',
      rating: 3.5,
      text: `Ты думаешь, что ты понял это, хотя на самом деле всего лишь думаешь, что ты понял то, о чем ты думаешь.<br/>
      Сколько бы иллюзий ты ни сочинил, они никогда не заполнят дыру в твоём сердце!<br/>
      Не будет трусостью отказаться от поставленной задачи. Трусостью будет не делать ничего.`,
      date: moment().add(-2, 'd'),
    }
  ];
  public static getReviews(): Promise<IReview[]> {
    return new Promise((resolve) => {
      setTimeout(() => resolve(ReviewsService.reviews), 2500);
    });
  }
}

export default ReviewsService;
