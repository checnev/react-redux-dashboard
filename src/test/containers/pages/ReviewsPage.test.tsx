import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { screen, waitFor } from '@testing-library/react'
import pretty from 'pretty';
import ReviewsPageContainer from 'containers/pages/Reviews';
import ReviewsService from 'services/ReviewsService';
import moment from 'moment';

const getReviews = jest.fn().mockResolvedValue([
  {
    id: 1,
    author: 'Fake Author',
    rating: 4.5,
    text: 'Review text',
    date: moment('10.12.2020', 'DD.MM.YYYY'),
  },
]);

ReviewsService.getReviews = getReviews;

let container: any = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
it('ReviewsPageContainer render', async () => {
  await act(async () => {
    render(<ReviewsPageContainer />, container);
  });

  await waitFor(() => {
    expect(screen.getByText('Fake Author')).toBeDefined();
  });
  
  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
