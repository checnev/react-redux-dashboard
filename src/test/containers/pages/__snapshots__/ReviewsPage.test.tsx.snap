// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsPageContainer render 1`] = `
"<div class=\\"MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-3\\">
  <div class=\\"MuiGrid-root makeStyles-header-2 MuiGrid-item MuiGrid-grid-xs-12\\">
    <h1 class=\\"MuiTypography-root MuiTypography-h4\\">Отзывы</h1>
  </div>
  <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6\\">
    <div class=\\"MuiPaper-root makeStyles-paper-5 MuiPaper-elevation1 MuiPaper-rounded\\">
      <div class=\\"MuiGrid-root makeStyles-root-4 MuiGrid-container MuiGrid-spacing-xs-3\\">
        <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\">
          <h4 class=\\"MuiTypography-root makeStyles-author-7 MuiTypography-h5\\">Fake Author</h4>
          <div><span class=\\"MuiRating-root MuiRating-sizeSmall MuiRating-readOnly\\" role=\\"img\\" aria-label=\\"4.5 Stars\\"><span class=\\"MuiRating-decimal\\"><span style=\\"width: 0%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span><span class=\\"MuiRating-decimal\\"><span style=\\"width: 0%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span><span class=\\"MuiRating-decimal\\"><span style=\\"width: 0%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span><span class=\\"MuiRating-decimal\\"><span style=\\"width: 0%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span><span class=\\"MuiRating-decimal\\"><span style=\\"width: 50%; overflow: hidden; z-index: 1; position: absolute;\\"><span class=\\"MuiRating-icon MuiRating-iconFilled\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span><span><span class=\\"MuiRating-icon MuiRating-iconEmpty\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeInherit\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\\"></path></svg></span></span></span></span></div>
        </div>
        <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6\\">
          <p class=\\"MuiTypography-root makeStyles-date-8 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignRight\\">10.12.2020</p>
        </div>
        <div class=\\"MuiGrid-root makeStyles-divider-9 MuiGrid-item MuiGrid-grid-xs-12\\">
          <hr class=\\"MuiDivider-root\\">
        </div>
        <div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\">
          <p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">Review text</p>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
